[Lim Jun Fun] iP

ljunfun opened the pull request.

ljunfun committed:
Reword Duke class to Stanley Class
Add features to printLine, sayHello, sayGoodbye

ljunfun committed:
Add echoWord to repeat user inputs

ljunfun committed:
Change Stanley class to Caifan class
Change echoWord to addTask
Add listTask function

ljunfun committed:
Add Task class
Add markTask, unmarkTask
Update listTask with status icon

lwenyi1 commented on src/main/java/Caifan.java:
```diff
@@ -0,0 +1,92 @@
+
+    public static void addTask(String line) {
+        printLine();
+        System.out.println("\tadded this for you my love: " + line);
```
```suggestion
        System.out.println("\tadded this for you my knee gar: " + line);
```

lwenyi1 commented on src/main/java/Caifan.java:
```diff
@@ -0,0 +1,92 @@
+    private static int taskCount = 0;
+
+    public static void printLine() {
+        String line = "\t____________________________________________________________";
```
Nice line.

lwenyi1 reviewed (commented):
Very cool. LGTM.
One suggestion made.

ljunfun committed:
Add Todo, Deadline, Event classes

NotGiggs commented on src/main/java/Caifan.java:
```diff
@@ -1,36 +1,63 @@
+        if (index >= 0 && index < taskCount) {
+            taskList[index].setDone(false);
+            printLine();
+            System.out.println("\tGet your lazy ass up and finish this ><");
```
Looks good.

NotGiggs commented on src/main/java/Caifan.java:
```diff
@@ -1,36 +1,63 @@
     public static void addTask(String line) {
         printLine();
-        System.out.println("\tadded this for you mi love: " + line);
+        System.out.println("\tadded this for you my love: " + line);
```
Good spot on the spelling mistake!

EdwinTun98 commented on src/main/java/Caifan.java:
```diff
@@ -0,0 +1,108 @@
+import java.util.Scanner;
+
```
could use more comments to explain the code

EdwinTun98 commented on src/main/java/Caifan.java:
```diff
@@ -0,0 +1,108 @@
+            throw new RuntimeException();
+        }
+    }
+
```
set task as complete and incomplete is clear

yeeern27 commented on src/main/java/Stanley.java:
```diff
@@ -0,0 +1,22 @@
+                "What can I do for you?";
+        System.out.println(welcomePhrase);
+    }
+    public static void sayGoodbye() {
```
It Is great putting these as functions so that it is easier to call when needed

yeeern27 commented on src/main/java/Stanley.java:
```diff
@@ -0,0 +1,22 @@
+        System.out.println(line);
+    }
+    public static void sayHello() {
+        String welcomePhrase =  "Hello! I'm Stanley!\n" +
```
name is good as it reflect what it means

EdwinTun98 commented on src/main/java/Caifan.java:
```diff
@@ -0,0 +1,108 @@
+                listTasks();
+            } else if (line.startsWith("mark")) {
+                int index = Integer.parseInt(line.substring(5)) - 1;
+                markTask(index);
```
could use a switch case instead of nested if loops

ljunfun replied:
implemented. thanks!

EdwinTun98 commented on src/main/java/Caifan.java:
```diff
@@ -0,0 +1,108 @@
+public class Caifan {
+    private static Task[] taskList = new Task[100];
+    private static int taskCount = 0;
+
```
good coding standard and clean code

ljunfun committed:
Change logic in main While loop to be switch case
Adds exceptions to handle errors with regards to each individual command
Adds functions to print messages whenever an exception is thrown
Adds InvalidDescriptionException class
Changed variable names for easier recognition

ljunfun committed:
Add TaskManager class to manage all functions previously implemented
Added exceptions to the various functions to catch
Fixed some bugs with exceptions thrown

ljunfun committed:
Merge branch 'branch-Level-5'

* branch-Level-5:
  Add TaskManager class to manage all functions previously implemented Added exceptions to the various functions to catch Fixed some bugs with exceptions thrown
  Change logic in main While loop to be switch case Adds exceptions to handle errors with regards to each individual command Adds functions to print messages whenever an exception is thrown Adds InvalidDescriptionException class Changed variable names for easier recognition

ljunfun committed:
Moves files into caifan package
Moves Deadline, Event, Task, and Todo classes into tasks package
Moves InvalidDescriptionException into exceptions package

ljunfun committed:
Merge branch 'branch-A-Packages'

* branch-A-Packages:
  Moves files into caifan package Moves Deadline, Event, Task, and Todo classes into tasks package Moves InvalidDescriptionException into exceptions package

ljunfun committed:
Changes Task class to a static class
Updates comments for readability

ljunfun committed:
Set up test cases for automated testing using the I/O redirection technique

ljunfun committed:
Adds new ArrayList as a new way to handle arrays
Adds handleDelete to handle the deletion of a task

ljunfun committed:
Creates Storage class to manage file management
Create functions within classes within task package to include functions to return important values

ljunfun committed:
Merge branch 'branch-Level-6'

* branch-Level-6:
  Adds new ArrayList as a new way to handle arrays Adds handleDelete to handle the deletion of a task

ljunfun committed:
Merge branch 'branch-Level-7'

* branch-Level-7:
  Creates Storage class to manage file management Create functions within classes within task package to include functions to return important values

ljunfun committed:
Fixes some bugs that appeared when trying to generate the file to store data

ljunfun committed:
Change storage logic so that it will store the list of task properly without duplicating
Adds new functions to Task class to obtain isDone status
Fixes some bugs

NCF3535 commented on src/main/java/caifan/Storage.java:
```diff
@@ -0,0 +1,112 @@
+import caifan.tasks.Todo;
+
+
+import java.io.*;
```
Imported classes should be listed explicitly, even if you are using the entire class.
For example:
`import java.util.List;`
This gives an excellent documentation value for the class at hand and makes the class easier to comprehend and maintain.

NCF3535 commented on src/main/java/caifan/Storage.java:
```diff
@@ -0,0 +1,112 @@
+                boolean isDone;
+                switch (splitInput[0].toUpperCase()) {
+                case "T":
+                    isDone = splitInput[1].equals("1");
```
There is code duplication here in the different cases:
`isDone = splitInput[1].equals("1");` and others
You could try to use the DRY (Don't Repeat Yourself) principle to avoid this from happening.

NCF3535 commented on src/main/java/caifan/Storage.java:
```diff
@@ -0,0 +1,112 @@
+            return type + " | " + (status ? "1" : "0") + " | " + description + " | " + ((Deadline)task).getDeadline();
+        case "E":
+            return type + " | " + (status ? "1" : "0") + " | " + description + " | " + ((Event)task).getStart() +
+                    " | " + ((Event)task).getEnd();
```
Try to break before the operator to improve readability of the program.

NCF3535 commented on src/main/java/caifan/Storage.java:
```diff
@@ -0,0 +1,112 @@
+
+    public static void createFile() {
+        File dir = new File(FILE_DIR);
+        File f = new File(FILE_PATH);
```
Use a better name to improve readability here instead of `dir` and `f`.

NCF3535 commented on src/main/java/caifan/TaskManager.java:
```diff
@@ -0,0 +1,265 @@
+        printLine();
+    }
+
+    /* print an error whenever an number is being entered that is
```
It is good that you have header comments to explain what the code does. However, do keep to Javadoc coding standard for header comments.

NCF3535 reviewed (commented):
Overall the code mostly conforms to coding standard, keep up the good work! Its nice to see that the chatbot has some personality to it too. Do note on the comments to improve readability of the code so that it is easier to maintain and comprehend.

ljunfun committed:
Amends the header lines to match the proper coding style

ljunfun committed:
Creates Parser, Ui, TaskList classes, and Command package to fit OOP Principles

ljunfun committed:
Merge branch 'A-MoreOOP'

* A-MoreOOP:
  Creates Parser, Ui, TaskList classes, and Command package to fit OOP Principles

ljunfun committed:
Implements find command

ljunfun committed:
Merge pull request #1 from ljunfun/branch-Level-9

Implements find command

ljunfun committed:
Merge branch 'master' into branch-A-JavaDoc

* master:
  Implements find command

ljunfun committed:
Adds JavaDoc to all relevant classes

ljunfun committed:
Merge pull request #2 from ljunfun/branch-A-JavaDoc

branch-A-JavaDoc

ljunfun committed:
Merge branch 'master' of https://github.com/ljunfun/ip

* 'master' of https://github.com/ljunfun/ip:
  Adds JavaDoc to all relevant classes

ljunfun committed:
Preliminary User Guide

ljunfun committed:
Adds README documentation
Fixes some Ui bugs and customises the chatbot

ljunfun committed:
Makes Ui nicer

ljunfun committed:
README update
